### PR TITLE
Fix access control issues with S3 API metadata directory

### PR DESCRIPTION
**Pull request by ZhuTopher**
Fixes an issue where non-root users in the Alluxio FS cannot create multipart uploads due to the S3 API metadata directory mode bits.

### CODE REVIEW

**Comment by ZhuTopher** [comment]
I tried to update the unit tests to include this, but it resulted in other errors I don't have time to deal with right now; they're on a different branch.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java [hunk 1]
```diff
@@ -173,8 +173,8 @@ public S3RestServiceHandler(@Context ServletContext context)
               .setRecursive(true)
               .setMode(PMode.newBuilder()
                   .setOwnerBits(Bits.ALL)
-                  .setGroupBits(Bits.READ_EXECUTE)
-                  .setOtherBits(Bits.READ_EXECUTE).build())
+                  .setGroupBits(Bits.ALL)
```
Comment (Xenorith) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:176]: unit test case possible?

Comment (ZhuTopher) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:176]: Yes, but no. The test cases are really... fidgety for some reason. https://github.com/Alluxio/alluxio/pull/16180#issuecomment-1244163136

I'll be getting the unit tests in afterwards

Comment (ZhuTopher) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:176]: Let me try again

Comment (Xenorith) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:176]: fine to follow up on a separate PR if this is blocking patch progress/deadline. just wanted to be sure the idea crossed your mind

Comment (Xenorith) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:176]: and i totally missed the PR level comment -> jumped right into the code diff

**Review by jenoudet — APPROVED** [review]

File: core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java (hunk 1 above)
Comment (ZhuTopher) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java:176]: Changed the resolution here; managed to fix the unit test issue I had by using `MUST_CACHE`.

File: tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java [hunk 2]
```diff
@@ -953,7 +955,11 @@ private void putObjectTest(String bucket, String objectKey, byte[] object, Strin
   }
 
   @Test
+  @Ignore
   public void testGetDeletedObject() throws Exception {
+    // This test requires the following property key change
+    // Configuration.set(PropertyKey.PROXY_S3_WRITE_TYPE, WriteType.CACHE_THROUGH.name());
+
```
Comment (ZhuTopher) [inline tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java:962]: @Xenorith I have to skip this unit test because it depends on UFS writes; which if enabled result in issues with the Authorization-related tests because this class uses a local UFS.

**Review by Xenorith — APPROVED** [review]

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Comment by ZhuTopher** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please
